#!/bin/sh -e
## 310_hppa-gas-unused.dpatch
##
## DP: Description: Fix build failure with gcc-4.3 -Werror on hppa.
## DP: Upstream status: cvs trunk

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p0 < $0;;
       -unpatch) patch $patch_opts -p0 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
--- gas/config/tc-hppa.c~	2007-07-03 13:01:04.000000000 +0200
+++ gas/config/tc-hppa.c	2008-05-21 22:21:27.000000000 +0200
@@ -5949,24 +5949,6 @@
   demand_empty_rest_of_line ();
 }
 
-/* Return TRUE if FRAG1 and FRAG2 are the same.  */
-
-static bfd_boolean
-is_same_frag (fragS *frag1, fragS *frag2)
-{
-
-  if (frag1 == NULL)
-    return FALSE;
-  else if (frag2 == NULL)
-    return FALSE;
-  else if (frag1 == frag2)
-    return TRUE;
-  else if (frag2->fr_type == rs_fill && frag2->fr_fix == 0)
-    return (is_same_frag (frag1, frag2->fr_next));
-  else
-    return FALSE;
-}
-
 #ifdef OBJ_ELF
 /* Build an entry in the UNWIND subspace from the given function
    attributes in CALL_INFO.  This is not needed for SOM as using
